### PR TITLE
[Improvement-16137][UI] Sub process support other project child node

### LINKED ISSUE #16162: [DSIP-48][Process] Sub process support other project child node
### Search before asking

- [X] I had searched in the [DSIP](https://github.com/apache/dolphinscheduler/issues/14102) and found no similar DSIP.

### Motivation

Currently, the sub-process task node can only select the current project workflow as a child node. We want to enable it to use workflows from different projects, allowing for flexible reuse instead of repeatedly creating definitions. We discovered that this feature can be achieved by simply modifying the frontend code.

### Design Detail

add a new form item call `child node project` in sub process node and save a new field data `childNodeProjectCode` in task params

### Compatibility, Deprecation, and Migration Plan

_No response_

### Test Plan

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yeahhhz**
## Purpose of the pull request

fix #16162 

## Verify this pull request

Screenshot 2024-06-12 at 13 11 26

### CODE REVIEW

**Comment by qingwli** [comment]
@songjianet  PTAL

File: dolphinscheduler-ui/src/locales/en_US/project.ts [hunk 1]
```diff
@@ -387,6 +388,9 @@ export default {
     task_type_tips: 'Please select a task type (required)',
     workflow_name: 'Workflow Name',
     workflow_name_tips: 'Please select a workflow (required)',
+    child_node_project: 'Child Node Project',
+    child_node_project_tips:
+      'Select this field for choose different project child node',
```
Comment (ruanwenjun) [inline dolphinscheduler-ui/src/locales/en_US/project.ts]: I would prefer to use Project rather than `Child Node Project`

Comment (yeahhhz) [reply dolphinscheduler-ui/src/locales/en_US/project.ts]: updated

File: dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-child-node.ts [hunk 2]
```diff
@@ -59,32 +63,79 @@ export function useChildNode({
     model.definition = res
   }
 
+  const getAccessibleProjectList = async () => {
+    queryProjectListPaging({
+      pageNo: 1,
+      pageSize: 999
+    })
```
Comment (caishunfeng) [inline dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-child-node.ts:70]: I think it's better to add api to query all projects for user.

File: dolphinscheduler-ui/src/views/projects/task/components/node/detail-modal.tsx [hunk 3]
```diff
@@ -194,14 +194,24 @@ const NodeDetailModal = defineComponent({
               ).then((res: any) => {
                 router.push({
                   name: 'workflow-instance-detail',
-                  params: { id: res.subProcessInstanceId },
+                  params: {
+                    id: res.subProcessInstanceId,
+                    projectCode:
+                      props.data.taskParams?.childNodeProjectCode ||
```
Comment (caishunfeng) [inline dolphinscheduler-ui/src/views/projects/task/components/node/detail-modal.tsx:200]:
```suggestion
                      props.data.taskParams?.subProjectCode ||
```

File: dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-child-node.ts (hunk 2 above)
Comment (yeahhhz) [reply dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-child-node.ts:70]: ok， will update later

**Review by songjianet — APPROVED** [review]

**Pull request closed without merging**